2-mandatory/3-shopping-list.js

```diff
@@ -18,10 +18,23 @@ let pantry = {
   cupboardContents: ["salt", "tinned tomatoes", "oregano"],
 };
 
+// write code here
 function createShoppingList(recipe) {
-  // write code here
+  let shoppingList = {};
+  shoppingList.name = recipe.name;
+  shoppingList.items = [];
+  recipe.ingredients.map(item => {
```

Comment:
Again, this code works, but what is the purpose of `Array.map`? How could you also use `Array.filter` to help you here?

Reply:
recipe.ingredients.filter((item) => 
    !pantry.fridgeContents.includes(item) &&
    !pantry.cupboardContents.includes(item))
    .forEach((item) => shoppingList.items.push(item));    // this way looks more smart

Reply:
This is an improvement, but can you use `.map` here instead of `.forEach`?

Reply:
How about:
```
shoppingList.items = recipe.ingredients
    .filter((item) =>
        !pantry.fridgeContents.includes(item) &&
        !pantry.cupboardContents.includes(item)
    );
```